presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -171,10 +171,12 @@ private ThriftMetastoreUtil() {}
 
     public static org.apache.hadoop.hive.metastore.api.Table toMetastoreApiTable(TableWithPrivileges table)
     {
-        return toMetastoreApiTable(table.getTable(), table.getPrincipalPrivileges());
+        org.apache.hadoop.hive.metastore.api.Table result = toMetastoreApiTableWithoutPrivileges(table.getTable());
+        result.setPrivileges(toMetastoreApiPrincipalPrivilegeSet(table.getPrincipalPrivileges()));
+        return result;
     }
 
-    public static org.apache.hadoop.hive.metastore.api.Table toMetastoreApiTable(Table table, PrincipalPrivileges privileges)
+    public static org.apache.hadoop.hive.metastore.api.Table toMetastoreApiTableWithoutPrivileges(Table table)
```

Comment:
We should remove this method since it is only used for tests (squash it into the method above).